cmd/fleet/vuln_process.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/WatchBeam/clock"
+	"github.com/fleetdm/fleet/v4/server/config"
+	"github.com/fleetdm/fleet/v4/server/datastore/mysql"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/spf13/cobra"
+)
+
+var dev bool
+
+func createVulnProcessingCmd(configManager config.Manager) *cobra.Command {
+	vulnProcessingCmd := &cobra.Command{
+		Use:   "vuln_processing",
+		Short: "Run the vulnerability processing features of Fleet",
+		Long: `The vuln_processing command is intended for advanced configurations that want to externally manage 
+vulnerability processing. By default the Fleet server command internally manages vulnerability processing via scheduled
+'cron' style jobs.`,
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := configManager.LoadConfig()
+			if dev {
+				applyDevFlags(&cfg)
+			}
+
+			ds, err := mysql.New(cfg.Mysql, clock.C)
+			if err != nil {
+				initFatal(err, "creating db connection")
+			}
+
+			// we need to ensure this command isn't running with an out-of-date database
+			status, err := ds.MigrationStatus(cmd.Context())
+			if err != nil {
+				initFatal(err, "retrieving migration status")
+			}
+
+			switch status.StatusCode {
+			case fleet.AllMigrationsCompleted:
+				// only continue if db is considered up-to-date
+				break
+			case fleet.NoMigrationsCompleted, fleet.SomeMigrationsCompleted, fleet.UnknownMigrations:
+				initFatal(errors.New("database migrations incompatible with current version"), "refusing to continue processing vulnerabilities")
+			}
+
+			logger := initLogger(cfg)
+			logger = kitlog.With(logger, fleet.CronVulnerabilities)
+
+			ctx := context.Background()
+			appConfig, err := ds.AppConfig(ctx)
+			if err != nil {
+				initFatal(err, "error fetching app config during vulnerability processing")
+			}
+			vulnConfig := cfg.Vulnerabilities
+			vulnPath := configureVulnPath(vulnConfig, appConfig, logger)
+			// this really shouldn't ever be empty string since it's defaulted, but could be due to some misconfiguration
+			// we'll throw an error here since the entire point of this command is to process vulnerabilities
+			if vulnPath == "" {
+				initFatal(errors.New("vuln path empty, check environment variables or app config yml"), "error during vulnerability processing")
+			}
```

Comment:
Before running, should we check if there are any entries in `locks` for vulnerabilities? (which would indicate if there are fleet instances configured to run vulnerabilities cron job).

Or maybe a warning? Though the solution of a leftover locks entry is currently to just log into the db and run `delete from locks where name = 'vulnerabilities';`...

Am ok with changes here, just trying to help the user not mess up and run multiple instances of vulnerabilities processing at the same time.

Reply:
I halfway considered still honoring the locking. In case multi tasks accidentally get launched, or even processing takes longer than the job run interval. Thoughts?

Reply:
Maybe it's safe to honor locks on the first iteration (better be on the safe side than allow concurrent vulnerability processing?).

If we decide to honor locks, then we'd need to use the `schedule` package (`server/service/schedule`) (it has capabilities to (1) lock+unlock when it has completed, (2) extend lock if the task hasn't finished and the interval is over).

Reply:
@zwass thoughts?